File: mentor-on-demand-app-client/src/app/models/mentor-search-result.ts
import { MentorSkill } from './mentor-skill';
import { Mentor } from './mentor';
import { Skill } from './skill';
import { MentorCalendar } from './mentor-calendar';

export class MentorSearchResult {
    
    private _mentor: Mentor;
    private _skill: Skill;
    private _mentorSkill: MentorSkill;
    private _mentorCalendars: MentorCalendar[];
    private _mentorFirstName: string;
    private _mentorLastName: string;
    private _contactNo: string;

    constructor(mentor: Mentor, skill: Skill, mentorSkill: MentorSkill, mentorCalendars: MentorCalendar[], mentorFirstName: string, mentorLastName: string, contactNo: string) {
        this._mentor = mentor;
        this._skill = skill;
        this._mentorSkill = mentorSkill;
        this._mentorCalendars = mentorCalendars;
        this._mentorFirstName = mentorFirstName;
        this._mentorLastName = mentorLastName;
        this._contactNo = contactNo;
    }

    get mentor(): Mentor {
        return this._mentor;
    }

    set mentor(mentor: Mentor) {
        this._mentor = mentor;
    }

    get skill(): Skill {
        return this._skill;
    }

    set skill(skill: Skill) {
        this._skill = skill;
    }

    get mentorSkill(): MentorSkill {
        return this._mentorSkill;
    }

    set mentorSkill(mentorSkill: MentorSkill) {
        this._mentorSkill = mentorSkill;
    }

    get mentorCalendars(): MentorCalendar[] {
        return this._mentorCalendars;
    }

    set mentorCalendars(mentorCalendars: MentorCalendar[]) {
        this._mentorCalendars = mentorCalendars;
    }

    get mentorFirstName(): string {
        return this._mentorFirstName;
    }

    set mentorFirstName(mentorFirstName: string) {
        this._mentorFirstName = mentorFirstName;
    }

    get mentorLastName(): string {
        return this._mentorLastName;
    }

    set mentorLastName(mentorLastName: string) {
        this._mentorLastName = mentorLastName;
    }

    get contactNo(): string {
        return this._contactNo;
    }

    set contactNo(contactNo: string) {
        this._contactNo = contactNo;
    }

}

File: mentor-on-demand-system-mentor-skill-ms/src/main/java/com/ibm/mods/mentorskill/model/MentorSearchResult.java
package com.ibm.mods.mentorskill.model;

import java.util.List;
import java.util.Map;

import com.ibm.mods.mentorskill.model.MentorSkill;

public class MentorSearchResult {

	private Mentor mentor;
	private Skill skill;
	private MentorSkill mentorSkill;
	private List<MentorCalendar> mentorCalendars;
	private String mentorFirstName;
	private String mentorLastName;
	private String contactNo;
	private Map<String, List<Training>> trainings;
	//private List<Training> trainings;
	
	public MentorSearchResult() {
		
	}
	
	public MentorSearchResult(Mentor mentor, Skill skill, MentorSkill mentorSkill, List<MentorCalendar> mentorCalendars, String firstname, String lastname, String contactNo, Map<String, List<Training>> trainings) {
		this.mentor = mentor;
		this.skill = skill;
		this.mentorSkill = mentorSkill;
		this.mentorCalendars = mentorCalendars;
		this.mentorFirstName = firstname;
		this.mentorLastName = lastname;
		this.contactNo = contactNo;
		this.trainings = trainings;
	}
	
	public Mentor getMentor() {
		return mentor;
	}

	public void setMentor(Mentor mentor) {
		this.mentor = mentor;
	}

	public Skill getSkill() {
		return skill;
	}

	public void setSkill(Skill skill) {
		this.skill = skill;
	}

	public MentorSkill getMentorSkill() {
		return mentorSkill;
	}

	public void setMentorSkill(MentorSkill mentorSkill) {
		this.mentorSkill = mentorSkill;
	}
	
	public List<MentorCalendar> getMentorCalendars() {
		return mentorCalendars;
	}

	public void setMentorCalendars(List<MentorCalendar> mentorCalendars) {
		this.mentorCalendars = mentorCalendars;
	}

	public String getMentorFirstName() {
		return mentorFirstName;
	}
	
	public void setMentorFirstName(String mentorFirstName) {
		this.mentorFirstName = mentorFirstName;
	}
	
	public String getMentorLastName() {
		return mentorLastName;
	}
	
	public void setMentorLastName(String mentorLastName) {
		this.mentorLastName = mentorLastName;
	}

	public String getContactNo() {
		return contactNo;
	}

	public void setContactNo(String contactNo) {
		this.contactNo = contactNo;
	}

	public Map<String, List<Training>>getTrainings() {
		return trainings;
	}

	public void setTrainings(Map<String, List<Training>> trainings) {
		this.trainings = trainings;
	}
	
}
